### PR TITLE
Guid helpers: extracting guids from dbs, parsing list strings

### PR DESCRIPTION
**Pull request by Dominik-Vogel**
I thought this code might be general enough to be of interest in qcodes. Adding two guids helpers:
- extract a dictionary of guids from all db files in the given folder and subfolders
- extract guids from a string representation of a list of strings.

@QCoDeS/core

### CODE REVIEW

**Review by astafan8 — COMMENTED** [review]
left a few suggestion, but in general, sure, let's get these in.

not sure if they belong to "guids" module as they don't really do any guid-related magic (say, parsing or generating them), so perhaps just a new module with some decent name?

so great to have the tests as well :)

File: qcodes/dataset/guids.py [hunk 1]
```diff
@@ -173,3 +178,68 @@ def validate_guid_format(guid: str) -> None:
...
+    Recursively find all db files under basepath and extract guids.
+
+    Args:
+        basepath: Path or str or directory where to search
+
+    Returns:
+        Tuple of Dictionary mapping paths to lists of guids as strings
+        and Dictionary mapping guids to db paths.
+    """
+    db_paths = Path(basepath).glob('**/*.db')
```
[the remark this answers is not in the document]
Comment (Dominik-Vogel) [reply qcodes/dataset/guids.py]: good point, thanks!

File: qcodes/dataset/guids.py [hunk 2]
```diff
@@ -173,3 +178,68 @@ def validate_guid_format(guid: str) -> None:
...
+        For an empty list/tuple/set or empty string an empty set is returned.
+    Examples:
+        >>> guids_from_str(
+        "['07fd7195-c51e-44d6-a085-fa8274cf00d6', \
+          '070d7195-c51e-44d6-a085-fa8274cf00d6']")
+        will return
+        ('07fd7195-c51e-44d6-a085-fa8274cf00d6',
+        '070d7195-c51e-44d6-a085-fa8274cf00d6')
+    """
+    guid = r"[\da-f]{8}-(?:[\da-f]{4}-){3}[\da-f]{12}"
```
[the remark this answers is not in the document]
Comment (Dominik-Vogel) [reply qcodes/dataset/guids.py]: yes, these are two good options, which I might have chosen, if I had to do this again. now that the regexp is in place it does multiple things at once and might be worth having: Check if it is actually only guids in the list and reading the list. One could also read the list as strings and then check if all items are guids with the regexp from the guids module. Personally now that this one works it does not seem worth the effort to change.

**Comment by Dominik-Vogel** [comment]
Thank you Mikhail, great comment. I moved the functions into `guid_helpers.py` not the best name, but I could not come up with a better one.

**Review by astafan8 — APPROVED** [review]
Undraft and lets merge, i think :)

**Pull request merged**